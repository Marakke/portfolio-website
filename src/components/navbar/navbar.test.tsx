import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Navbar from "./index";

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  navbar: {
    items: [
      {
        name: "Item1"
      },
      {
        name: "Item2"
      },
    ]
  }
}

describe("Navbar", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Navbar component correctly", () => {
    const { container, getByTestId } = render(<Navbar />);
    expect(container).toMatchSnapshot();

    const navbarContainer = getByTestId("navbar-container");
    expect(navbarContainer).toBeInTheDocument();

    const anchorElements = container.querySelectorAll("a");
    expect(anchorElements.length).toEqual(2);

    expect(anchorElements[0]).toHaveTextContent("Item1");
    expect(anchorElements[1]).toHaveTextContent("Item2");

    expect(anchorElements[0].href).toBe("http://localhost/#item1");
    expect(anchorElements[1].href).toBe("http://localhost/#item2");
  });
});
