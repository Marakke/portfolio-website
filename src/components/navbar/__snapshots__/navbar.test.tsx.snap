// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders the Navbar component correctly 1`] = `
<div>
  <div
    class="navbar-container"
    data-testid="navbar-container"
  >
    <a
      class="navbar-item"
      href="#item1"
    >
      Item1
    </a>
    <a
      class="navbar-item"
      href="#item2"
    >
      Item2
    </a>
  </div>
</div>
`;
